app/src/main/java/org/mozilla/fenix/home/collections/TabInCollection.kt

```diff
@@ -0,0 +1,300 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.material.DismissDirection
+import androidx.compose.material.DismissDirection.EndToStart
+import androidx.compose.material.DismissDirection.StartToEnd
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.Icon
+import androidx.compose.material.SwipeToDismiss
+import androidx.compose.material.rememberDismissState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.draw.drawWithContent
+import androidx.compose.ui.graphics.drawscope.clipRect
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.dp
+import mozilla.components.browser.state.state.recover.RecoverableTab
+import mozilla.components.concept.engine.Engine
+import mozilla.components.feature.tab.collections.Tab
+import org.mozilla.fenix.R.drawable
+import org.mozilla.fenix.R.string
+import org.mozilla.fenix.compose.inComposePreview
+import org.mozilla.fenix.compose.list.FaviconListItem
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.toShortUrl
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Rectangular shape with only right angles used to display a middle tab.
+ */
+private val middleTabShape = RoundedCornerShape(0.dp)
+
+/**
+ * Rectangular shape with only the bottom corners rounded used to display the last tab in a collection.
+ */
+private val bottomTabShape = RoundedCornerShape(bottomStart = 8.dp, bottomEnd = 8.dp)
+
+/**
+ * Composable used to display an individual [Tab].
+ *
+ * @param tab [Tab] to display.
+ * @param isLastInCollection Whether the tab is to be shown between others or as the last one in collection.
+ * @param onClick Invoked when the user click on the tab.
+ * @param onRemove Invoked when the user removes the tab informing also if the tab was swiped to be removed.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun TabInCollection(
+    tab: Tab,
+    isLastInCollection: Boolean,
+    onClick: () -> Unit,
+    onRemove: (Boolean) -> Unit,
+) {
+    val dismissState = rememberDismissState()
+
+    if (dismissState.isDismissed(StartToEnd) || dismissState.isDismissed(EndToStart)) {
+        onRemove(true)
+    }
+
+    SwipeToDismiss(
```

Comment:
I am curious if you used any particular resources/docs for this. Just want to be looking at the same resources while reviewing this.

Reply:
There the [official documentation](https://developer.android.com/reference/kotlin/androidx/compose/material/package-summary#SwipeToDismiss(androidx.compose.material.DismissState,androidx.compose.ui.Modifier,kotlin.collections.Set,kotlin.Function1,kotlin.Function1,kotlin.Function1)) based on which I tried to obtain the exact same result as we have now. (The delete icons from the background which appear selectively based on the swipe direction needed special attention)